scheduler/docs/configuration.adoc

```diff
@@ -78,7 +78,14 @@ If need it to bind to another port, you can specify that with the `:local-port`
   The value of this key is a set of usernames who should be considered administrators when using the `configfile-admins-auth` authorization-fn.
 
 `:rate-limit`::
-  Configure rate limits for the scheduler. `rate-limit` is a map with two possible keys, `user-limit-per-m`. `user-limit-per-m` is the max requests a single user can send in a minute. The default is 600 requests.
+  Configure rate limits for the scheduler. `rate-limit` is a map with three possible keys `:user-limit-per-m` , `:job-submission`, `:expire-minutes`. These keys control rate limits.
+
+ * `:user-limit-per-m` is the max REST requests a single user can send in a minute. The default is 600 requests.
+
+ * `:job-submission` is a dictionary with a _token bucket filter configuration_ that allows per-user restriction of the job creation rate.
+ * `:expire-minutes` For our token-bucket-filter rate limits, we will create one token bucket filter rate limit object for each user. If a user has become very idle, we expire old unused rate-limit entries after a time period. This should be set to several hours, or at least as high as (`:bucket-size`/`:tokens-replenished-per-minute`)
```

Comment:
Should we throw an error if the expire minutes are set to lower than bucket size / tokens replenished per minute?

Reply:
This is a component of an explanation of why its OK to give users full buckets when we initialize a new bucket for a never-before-seen-user. Given that the new code can handle 1000 submission/second, and our planned rate limit of 100 submisisons/second (i.e., 6k/minute). A bucketsize of 20k jobs, has a threshold of 4 minutes. Any sane configured expire timeout is going to be more than 4 minutes.

Reply:
Sure, but Cook operators other than us may need to use different bucket sizes and/or token replenish rates. It seems that if it's worth mentioning in the docs that it should be at least as high as a certain number, it may be worth simply checking for that in the code and raising an error.